src/hivpy/pregnancy.py

```diff
@@ -0,0 +1,268 @@
+                                            & (~pop.data[col.LOW_FERTILITY])
+                                            & (~pop.data[col.PREGNANT])
+                                            & (pop.data[col.NUM_CHILDREN] < self.max_children)
+                                            & ((pop.data[col.NUM_PARTNERS] > 0)
+                                                | pop.data[col.LONG_TERM_PARTNER])]
+
+        # continue if sexually active females are present this timestep
+        if len(active_female_population) > 0:
+
+            # at least 6 months must pass since last childbirth
+            # (15 months since last pregnancy start date)
+            previously_pregnant = active_female_population[active_female_population[col.LAST_PREGNANCY_DATE].notnull()]
+            pregnancy_ready = ~active_female_population[col.LOW_FERTILITY]
+            if len(previously_pregnant) > 0:
+                # mask
+                pregnancy_ready = pregnancy_ready & (previously_pregnant[col.LAST_PREGNANCY_DATE]
+                                                     + timedelta(days=450) <= pop.date)
+
+            # group females by age groups
+            age_groups = np.digitize(pop.data.loc[active_female_population[pregnancy_ready].index, col.AGE],
+                                     [15, 25, 35, 45, 55])
+            # TODO: change age group col name to be more descriptive
+            pop.data.loc[active_female_population[pregnancy_ready].index, col.AGE_GROUP] = age_groups
+            # calculate pregnancy outcomes
+            pregnancy = pop.transform_group([col.AGE_GROUP, col.LONG_TERM_PARTNER,
+                                             col.NUM_PARTNERS, col.WANT_NO_CHILDREN],
+                                            self.calc_preg_outcomes,
+                                            sub_pop=active_female_population[pregnancy_ready].index)
+            # assign outcomes
+            pop.data.loc[active_female_population[pregnancy_ready].index, col.PREGNANT] = pregnancy
+            # use pregnancy outcomes as mask to assign current date as pregnancy date
+            pop.data.loc[active_female_population[pregnancy_ready].index[pregnancy], col.LAST_PREGNANCY_DATE] = pop.date
+
+        self.update_antenatal_care_testing(pop)
+        self.update_births(pop)
+        self.update_want_no_children(pop)
+
+    def update_antenatal_care_testing(self, pop):
+        """
+        Determine who is in antenatal care and receiving
+        prevention of mother to child transmission care.
+        """
+        # get pregnant population
+        pregnant_population = pop.data[pop.data[col.PREGNANT]]
+        # update probability of antenatal care attendance
+        self.prob_anc = min(max(self.prob_anc, 0.1) + self.rate_testanc_inc, 0.975)
+
+        # anc outcomes
+        r = rng.uniform(size=len(pregnant_population))
+        anc = r < self.prob_anc
+        pop.data.loc[pregnant_population.index, col.ANC] = anc
```

Comment:
This ANC state is persistant even after pregnancy is ended. Is this correct? Should ANC in one pregnancy imply ANC for all future pregnancies?